Data/models/DataController.swift

```diff
@@ -30,22 +31,75 @@ public class DataController: NSObject {
-                    try migrationContainer.persistentStoreCoordinator.migratePersistentStore(oldStore, to: storeURL, options: nil, withType: NSSQLiteStoreType)
-                    try migrationContainer.persistentStoreCoordinator.destroyPersistentStore(at: oldStoreURL, ofType: NSSQLiteStoreType, options: nil)
-                    try sqliteFiles(from: oldStoreURL.deletingLastPathComponent(), dbName: DataController.databaseName).forEach(FileManager.default.removeItem)
-                }
-            } catch {
-                log.error(error)
-            }
+        
+        // This logic must account for 4 different situations:
+        // 1. New Users (no migration, use new location
+        // 2. Upgraded users with successful migration (use new database)
+        // 3. Upgraded users with unsuccessful migrations (use new database, ignore old files)
+        // 4. Upgrading users (attempt migration, if fail, use old store, if successful delete old files)
+        //      - re-attempt migration on every new app version, until they are in #2
+        
+        let specificStoreExists: (URL) -> Bool = { url in
+            FileManager.default.fileExists(atPath: url.path)
+        }
+        
+        if !specificStoreExists(oldDocumentStoreURL) || specificStoreExists(supportStoreURL) {
+            // Old store absent, no data to migrate (#1 | #3)
+            // or
+            // New store already exists, do not attempt to overwrite (#2)
+            
+            // Update flag to avoid re-running this logic
+            Preferences.Database.DocumentToSupportDirectoryMigration.completed.value = true
+            return
+        }
+        
+        // Going to attempt migration (#4 in some level)
+        
+        let coordinator = migrationContainer.persistentStoreCoordinator
+
+        guard let oldStore = coordinator.persistentStore(for: oldDocumentStoreURL) else {
+            throw MigrationError.OldStoreMissing("Old store unavailable")
+        }
+        
+        // Attempting actual database migration Document -> Support 🤞
+        do {
+            let migrationOptions = [
+                NSPersistentStoreFileProtectionKey: true
+            ]
+            try coordinator.migratePersistentStore(oldStore, to: supportStoreURL, options: migrationOptions, withType: NSSQLiteStoreType)
+        } catch {
+            throw MigrationError.MigrationFailed("Document -> Support database migration failed: \(error)")
+            // Migration failed somehow, and old store is present. Flag not being updated 😭
         }
+        
+        // Regardless of cleanup logic, the actual migration was successful, so we're just going for it 🙀😎
+        Preferences.Database.DocumentToSupportDirectoryMigration.completed.value = true
+        
+        // Cleanup time 🧹
+        do {
+            try coordinator.destroyPersistentStore(at: oldDocumentStoreURL, ofType: NSSQLiteStoreType, options: nil)
```

Comment:
Don't we need `NSPersistentStoreFileProtectionKey` option here as well?

Reply:
Based on my testing, I'm not even entirely sure that's a valid key for `destroy`. We considered just excluding this entirely, as we're deleting the files right after, so could do that.